test/unit/action/controller_tests.rb

```diff
@@ -0,0 +1,132 @@
+require "assert"
+require "much-rails/action/controller"
+
+require "test/support/actions/show"
+require "test/support/fake_action_controller"
+
+module MuchRails::Action::Controller
+  class UnitTests < Assert::Context
+    desc "MuchRails::Action::Controller"
+    subject { unit_module }
+
+    let(:unit_module) { MuchRails::Action::Controller }
+
+    should "include MuchRails::Plugin" do
+      assert_that(subject).includes(MuchRails::Plugin)
+    end
+  end
+
+  class ReceiverTests < UnitTests
+    desc "receiver"
+    subject { receiver_class }
+
+    let(:receiver_class) {
+      Class.new { include FakeActionController }
+    }
+  end
+
+  class ReceiverInitTests < ReceiverTests
+    desc "when init"
+    subject { receiver_class.new(params1) }
+
+    let(:params1) {
+      {
+        MuchRails::Action::Router::ACTION_CLASS_PARAM_NAME => "Actions::Show",
+        controller: "actions",
+        action: "show",
+        nested: {
+          value: "VALUE 1"
+        },
+      }
+    }
+
+    should have_readers :much_rails_action_class
+
+    should have_imeths MuchRails::Action::Router::CONTROLLER_METHOD_NAME
+    should have_imeths :much_rails_action_class_name
+    should have_imeths :much_rails_action_params
+    should have_imeths :require_much_rails_action_class
+    should have_imeths :permit_all_much_rails_action_params
+
+    should "know its attributes" do
+      assert_that(subject.much_rails_action_class_name)
+        .equals("::Actions::Show")
+
+      assert_that(subject.much_rails_action_params).equals(
+        nested: { value: "VALUE 1" },
+        value: "VALUE 1",
+      )
```

Comment:
```suggestion
      assert_that(subject.much_rails_action_params)
        .equals(
          nested: { value: "VALUE 1" },
          value: "VALUE 1",
        )
```
I believe we're doing this now.

Reply:
thanks!